### ISSUE TITLE
Add braces to multiline if-statements

### ISSUE THREAD

**Issue by 14mr0n** [scalameta/scalafmt#1463]
I suggest to have an option, to rewrite multiline if-statements to always use curly braces.
Single line should be left untouched in general or optionally formatted.

I used that style before, as it is supported by IntelliJ and I am missing this in scalafmt.
A good discussion, if anybody really needs to get convinced, can be found on [stackoverflow](https://stackoverflow.com/a/2125078/1787809).
There is even a real world example of an apple bug from 2014.

And scalastyle supports [this](http://www.scalastyle.org/rules-1.0.0.html#org_scalastyle_scalariform_IfBraceChecker) too.
They gave the option to make an exception for double lines. 
I think this is already to dangerous and should force braces too.

If you turn this option on in IntelliJ, the editor automatically inserts braces if you enter a newline after the if conditions parentheses are closed.
Is it possible to do the same if the scalafmt formatter is selected in IntelliJ?

## Example
So this code
```
if(shouldDoThis)
  doThis()
else
  doThat()
```

becomes
```
if(shouldDoThis){
  doThis()
} else {
  doThat()
}
```

**Comment by tmnd1991** [comment]
I might have time to give a try to this feature implementation but I have almost no familiarity with the code base. Is this a good first issue?

**Comment by poslegm** [comment]
@tmnd1991 Hello!

This is quite a difficult task. It should be done as rewrite rule, you can see how the other rule is implemented - [RedundantBraces](https://github.com/scalameta/scalafmt/blob/master/scalafmt-core/shared/src/main/scala/org/scalafmt/rewrite/RedundantBraces.scala) for example. I am not sure that all required logic can be encapsulated in rewrite rule only, but it's a good point to start. Feel free to ask any questions here or in scalafmt discord.

**Comment by tmnd1991** [comment]
I don't get why I can log into the discord channel, so I'll ask the question here. If I try to start trying to add a "similar" feature to the redundantBraces rewrite rule? My purpose is a new setting of the rule: `removeFromSingleLineIf` true by default which if set to false, makes the rule do not rewrite if statements. For example:

```scala
if (cond) {
  doSomething()
} else {
  doSomethingElse()
}
```

will not be rewritten as:

```scala
if (cond)
  doSomething()
else
  doSomethingElse()
```

**Comment by poslegm** [comment]
I would prefer to see this as a separate rule because the next logical step is ability to always _add_ braces:

```scala
<<<
if (cond)
  doSomething()
else
  doSomethingElse()
>>>
if (cond) {
  doSomething()
} else {
  doSomethingElse()
}
```

So I would make a rule `SingleExpressionIfBraces` with setting `rewrite.singleExpressionIfBraces.action = add 
| remove`. But configuration naming is open question.

**Comment by kitbellew** [comment]
i'd say this should be done in `FormatWriter`, as that's the only place where it's known whether the code is formatted with single line or multiple.

**Comment by kitbellew** [comment]
btw, this has the same problem as #1222:
```
val a =
  if (cond)
    7
      -5
  else
   2
     -0
```
cannot be rewritten as
```
val a =
  if (cond) {
    7
      -5
  } else {
   2
     -0
  }
```
i'd recommend giving up on this :)

**Comment by kitbellew** [comment]
should be covered by #2019

**Comment by tmnd1991** [comment]
Good to know ❤️

**Comment by kitbellew** [comment]
> Good to know ❤️

it's not implemented, though, but @vito-c said he was working on it.

**Comment by tmnd1991** [comment]
Yeah, actually digging that post made me figure out that there are “options” in RedundantBraces right now. Between them there is the ability to disable it on if/else, and used in conjunction with scalastyle rule that checks for if/else always having braces, it’s a semi-automatic way to have this feature in my code based!

**Comment by vito-c** [comment]
> it's not implemented, though, but @vito-c said he was working on it.

slowly but surely :) I got torn away from it for a bit.
